google_nest_sdm/webrtc_util.py

```diff
@@ -1,114 +1,108 @@
-# SDP Direction Constants
-DIRECTION_SENDRECV = 'sendrecv'
-DIRECTION_SENDONLY = 'sendonly'
-DIRECTION_RECVONLY = 'recvonly'
-DIRECTION_INACTIVE = 'inactive'
+"""Library with functions for manipulating WebRTC requests/responses."""
+from enum import StrEnum
+
+# SDP direction constants
+class SDPDirection(StrEnum):
+    SENDRECV = "sendrecv"
+    SENDONLY = "sendonly"
+    RECVONLY = "recvonly"
+    INACTIVE = "inactive"
+
+# SDP media kind constants
+class SDPMediaKind(StrEnum):
+    AUDIO = "audio"
+    VIDEO = "video"
+    APPLICATION = "application"
+
+def _get_media_direction(sdp: str, kind: SDPMediaKind) -> SDPDirection | None:
+    """
+    Retrieves the direction of media tracks from the SDP based on the kind (audio/video).
+
+    Args:
+        sdp (str): The SDP content
+        kind (SDPMediaKind): The kind of media track to check ('audio' or 'video').
 
-AVAILABLE_DIRECTIONS = [DIRECTION_SENDRECV,
-                        DIRECTION_SENDONLY,
-                        DIRECTION_RECVONLY,
-                        DIRECTION_INACTIVE]
+    Returns:
+        SDPMediaKind: The direction of the media track. One of 'sendrecv', 'sendonly', 'recvonly', or 'inactive'.
+    """
+    # Track if we are in the desired media section
+    in_media_section = False
+
+    for line in sdp.splitlines():
+        # Check if the line is a media description line
+        if line.startswith("m="):
+            in_media_section = line.startswith(f"m={kind}")
+        # If we're in the desired media section, check for direction
+        if in_media_section and line.startswith("a="):
+            for direction in SDPDirection:
+                if line.startswith(f"a={direction}"):
+                    return direction
+    return None
+
+
+def _update_direction_in_answer(answer_sdp: str, kind: SDPMediaKind, old_direction: SDPDirection, new_direction: SDPDirection) -> str:
+    """
+    Updates the direction of a specific media track in the SDP answer if it matches a certain direction.
 
+    Args:
+        answer_sdp (str): The SDP answer in string format.
+        kind (SDPMediaKind): The kind of media track to update ('audio' or 'video').
+        old_direction (SDPDirection): The old direction to find ('sendrecv', 'sendonly', 'recvonly', 'inactive').
+        new_direction (SDPDirection): The new direction to set ('sendrecv', 'sendonly', 'recvonly', 'inactive').
 
-class WebRTCUtility:
+    Returns:
+        str: The updated SDP with the new direction.
     """
-    This class provides methods for managing WebRTC SDP.
+
+    # Update the SDP
+    updated_sdp_lines = []
+    in_media_section = False
+    for line in answer_sdp.split("\r\n"):
+        if line.startswith("m="):
+            in_media_section = line.startswith(f"m={kind}")
+        if in_media_section and line.startswith("a="):
+            # Update the direction line if it matches the kind
+            if line.startswith(f"a={old_direction}"):
```

Comment:
Is there any suffix of this line that needs to be preserved? or should this only match an entire line? Or is it ok to drop the suffix?